### PR TITLE
Update Automation DSC Template

### PR DESCRIPTION
**Pull request by bobbytreed**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [X ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

**Comment by bobbytreed** [comment]
@mgreenegit Let me know what you think

File: 101-automation-configuration/azuredeploy.json [hunk 1]
```diff
@@ -1,120 +1,150 @@
...
       "defaultValue": ""
     },
     "compileName": {
-      "type": "string",
+      "type"    : "string",
       "metadata": {
         "description": "Unique value to identify compilation job"
       },
-      "defaultValue": "[guid(resourceGroup().id, deployment().name)]"
+      "defaultValue": "[guid(resourceGroup().id, deployment().name, utcNow())]"
```
Comment (bmoore-msft) [inline 101-automation-configuration/azuredeploy.json]: just use:

"defaultValue": "[newGuid()]"

File: 101-automation-configuration/azuredeploy.json [hunk 2]
```diff
@@ -1,120 +1,150 @@
...
-      "defaultValue": "contoso.local"
-    },
     "_artifactsLocation": {
-      "type": "string",
+      "type"    : "string",
       "metadata": {
         "description": "Path to the nested templates used in this deployment"
       },
-      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-automation-configuration/"
+      "defaultValue": "https://raw.githubusercontent.com/bobbytreed/azure-quickstart-templates/bobbytreed-dsc/101-automation-configuration/"
```
Comment (bmoore-msft) [inline 101-automation-configuration/azuredeploy.json]: revert this change

File: 101-automation-configuration/azuredeploy.json [hunk 3]
```diff
@@ -1,120 +1,150 @@
...
+    "provisionKeyVaultUrl"     : "[uri(parameters('_artifactsLocation'), concat('nested/provisionKeyVault.json', parameters('_artifactsLocationSasToken')))]"
   },
   "resources": [
     {
-      "name": "provisionConfiguration",
-      "type": "Microsoft.Resources/deployments",
-      "apiVersion": "2018-02-01",
+      "name"      : "provisionKeyVault",
+      "type"      : "Microsoft.Resources/deployments",
+      "apiVersion": "2016-09-01",
```
Comment (bmoore-msft) [inline 101-automation-configuration/azuredeploy.json]: update apiVersion - see test results for all instances of this...

File: 101-automation-configuration/azuredeploy.json [hunk 4]
```diff
@@ -217,28 +248,36 @@
             "value": "[parameters('adminUsername')]"
           },
           "adminPassword": {
-            "value": "[parameters('adminPassword')]"
+            "reference": {
+              "keyVault": {
+                "id": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().Name, '/providers/Microsoft.KeyVault/vaults/', variables('keyVaultName'))]"
```
Comment (bmoore-msft) [inline 101-automation-configuration/azuredeploy.json]: remove concat() and use resourceId

File: 101-automation-configuration/azuredeploy.json [hunk 5]
```diff
@@ -217,28 +248,36 @@
...
+              "secretName": "VMPassword"
+            }
           },
           "NicName": {
             "value": "[variables('NicName')]"
           },
-          "automationAccountName": {
-            "value": "[variables('automationAccountName')]"
+          "registrationUrl": {
+            "value": "[reference('provisionConfiguration', '2016-09-01').outputs.registrationUrl.value]"
```
Comment (bmoore-msft) [inline 101-automation-configuration/azuredeploy.json]: update apiVersion

File: 101-automation-configuration/azuredeploy.json [hunk 6]
```diff
@@ -217,28 +248,36 @@
...
+          },
+          "registrationKey": {
+            "value": "[reference('provisionConfiguration', '2016-09-01').outputs.registrationKey.value]"
```
Comment (bmoore-msft) [inline 101-automation-configuration/azuredeploy.json]: update apiVersion

File: 101-automation-configuration/dscConfigurations/DSCFromGPO.ps1 [hunk 7]
```diff
@@ -0,0 +1,6501 @@
...
+              ValueType = 'Dword'
+              Key = 'HKLM:\Software\Policies\Microsoft\Windows\CurrentVersion\Internet Settings\Zones\3'
+              ValueData = 3
+
+         }
+
+         Registry 'Registry(POL): HKLM:\Software\Policies\Microsoft\Windows\CurrentVersion\Internet Settings\Zones\3\2004'
+         {
+              ValueName = '2004'
+              V
```
Comment (bmoore-msft) [inline 101-automation-configuration/dscConfigurations/DSCFromGPO.ps1]: remove commented code

File: 101-automation-configuration/azuredeploy.json [hunk 8]
```diff
@@ -1,120 +1,150 @@
...
+      "type"    : "string",
       "metadata": {
         "description": "Path to the nested templates used in this deployment"
       },
-      "defaultValue": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-automation-configuration/"
+      "defaultValue": "https://raw.githubusercontent.com/bobbytreed/azure-quickstart-templates/bobbytreed-dsc/101-automation-configuration/"
     },
     "_artifactsLocationSasToken": {
-      "type": "string",
+      "type"    : "string",
```
Comment (bmoore-msft) [inline 101-automation-configuration/azuredeploy.json]: this must be "secureString"

File: 101-automation-configuration/nested/provisionConfiguration.json [hunk 9]
```diff
@@ -23,141 +23,117 @@
...
       "type": "securestring"
     },
-    "domainName": {
-      "type": "string"
-    },
     "compileName": {
-      "type": "string",
-      "defaultValue": "[guid(resourceGroup().id, deployment().name)]"
+      "type"        : "string",
+      "defaultValue": "[guid(resourceGroup().id, deployment().name, utcNow())]"
```
Comment (bmoore-msft) [inline 101-automation-configuration/nested/provisionConfiguration.json]: use newGuid()

File: 101-automation-configuration/nested/provisionKeyVault.json [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+        "type"      : "Microsoft.KeyVault/vaults",
+        "name"      : "[parameters('keyVaultName')]",
+        "apiVersion": "2018-02-14",
+        "location"  : "[parameters('location')]",
+        "properties": {
+          "enabledForDeployment"        : "[parameters('enabledForDeployment')]",
+          "enabledForDiskEncryption"    : "[parameters('enabledForDiskEncryption')]",
+          "enabledForTemplateDeployment": "[parameters('enabledForTemplateDeployment')]",
+          "tenantId"                    : "[parameters('tenantId')]",
+          "accessPolicies"              : [],
```
Comment (bmoore-msft) [inline 101-automation-configuration/nested/provisionKeyVault.json]: remove empty properties

Comment (bobbytreed) [reply 101-automation-configuration/nested/provisionKeyVault.json]: https://github.com/Azure/azure-quickstart-templates/issues/5517

Errors if you remove it.

File: 101-automation-configuration/azuredeploy.json (hunk 5 above)
Comment (bobbytreed) [reply 101-automation-configuration/azuredeploy.json]: AutomationAccount's newest api is from 2015

**Comment by bobbytreed** [comment]
#sign-off

File: 101-automation-configuration/azuredeploy.json [hunk 11]
```diff
@@ -217,28 +251,36 @@
             "value": "[parameters('adminUsername')]"
           },
           "adminPassword": {
-            "value": "[parameters('adminPassword')]"
+            "reference": {
+              "keyVault": {
+                "id": "[resourceId(resourceGroup().Name, 'Microsoft.KeyVault/vaults', variables('keyVaultName'))]"
```
Comment (bmoore-msft) [inline 101-automation-configuration/azuredeploy.json]: remove resourceGroup().name

File: 101-automation-configuration/azuredeploy.json (hunk 5 above)
Comment (bmoore-msft) [reply 101-automation-configuration/azuredeploy.json]: you're referencing Microsoft.Resources/deployments

File: 101-automation-configuration/nested/provisionConfiguration.json [hunk 12]
```diff
@@ -23,141 +23,127 @@
...
       ]
     }
   ],
-  "outputs": {}
-}
\ No newline at end of file
+  "outputs": {
+    "registrationUrl": {
+       "type" : "string",
+       "value": "[reference(concat('Microsoft.Automation/automationAccounts/', parameters('automationAccountName'))).registrationUrl]"
```
Comment (bmoore-msft) [inline 101-automation-configuration/nested/provisionConfiguration.json]: remove concat() and use resourceId()

File: 101-automation-configuration/nested/provisionConfiguration.json [hunk 13]
```diff
@@ -23,141 +23,127 @@
...
+    },
+   "registrationKey": {
+     "type" : "string",
+     "value": "[listKeys(resourceId('Microsoft.Automation/automationAccounts/', parameters('AutomationAccountName')), '2018-01-15').Keys[0].value]"
```
Comment (bmoore-msft) [inline 101-automation-configuration/nested/provisionConfiguration.json]: update apiVersion

File: 101-automation-configuration/nested/provisionKeyVault.json [hunk 14]
```diff
@@ -0,0 +1,119 @@
...
+    "parameters"    : {
+      "keyVaultName": {
+        "type"    : "string",
+        "metadata": {
+          "description": "Specifies the name of the key vault."
+        }
+      },
+      "location": {
+        "type"        : "string",
+        "defaultValue": "[resourceGroup().location]",
```
Comment (bmoore-msft) [inline 101-automation-configuration/nested/provisionKeyVault.json]: remove defaultValue since this must be passed through

File: 101-automation-configuration/nested/provisionConfiguration.json [hunk 15]
```diff
@@ -23,141 +23,127 @@
...
   "resources": [
     {
-      "name": "[parameters('automationAccountName')]",
-      "type": "Microsoft.Automation/automationAccounts",
-      "apiversion": "2018-01-15",
-      "location": "[parameters('location')]",
-      "comments": "Automation account for OMS",
+      "name"      : "[parameters('automationAccountName')]",
+      "type"      : "Microsoft.Automation/automationAccounts",
+      "apiVersion": "2015-10-31",
```
Comment (bmoore-msft) [inline 101-automation-configuration/nested/provisionConfiguration.json]: update apiVersion here and below

File: 101-automation-configuration/nested/provisionConfiguration.json [hunk 16]
```diff
@@ -23,141 +23,127 @@
...
-          "name": "[variables('dscResources').xActiveDirectory.name]",
-          "type": "modules",
-          "apiVersion": "2018-01-15",
-          "location": "[parameters('location')]",
-          "dependsOn": [
-            "[parameters('AutomationAccountName')]"
+          "name"      : "[variables('dscResources').SecurityPolicyDSC.name]",
+          "type"      : "modules",
+          "apiVersion": "2015-10-31",
+          "location"  : "[resourceGroup().location]",
```
Comment (bmoore-msft) [inline 101-automation-configuration/nested/provisionConfiguration.json]: all location properties must be parameters('location')

**Comment by bmoore-msft** [comment]
Closing abandoned request - please reopen if you can address the feedback.
